doc/adr/0018-synchronising-queries-across-indexers-running-in-parallel.rst

```diff
@@ -0,0 +1,65 @@
+
+This solution also forgoes the possibility of splitting up the indexers into different processes or machines. Our customers complained about the excessive resource consumption, performance and the inability to easily add their own indexers to the chain index. Giving users the possibility of distributing their indexers will partially address the first concern by splitting resource consumption on several machines (which is also addressed at the indexer design level by an inherently scalable design) and indirectly the second (since indexers are distributable, they have a simple, standard API that allows any customer to roll their own version, implicitly solving the hard problems like rollbacks and synchronisation).
+
+Decision
+--------
+
+1. Change the `Request` type to include a preferred slot number and how many results we want to receive. If there is no preferred slot number, the top available slot number is assumed.
+2. Change the `Response` type to include all the requested query results encoded as the full query result, followed by a set od deltas (events).
+3. [Optional] Add a new function to the `Queryable` type class that witnesses the folding of Events into the query result.
+
+Example:
+
+Give a request for slot number 10, with a maximum of 5 results, some of the indexers might output:
+
+Indexer 01: State at slot 6, Delta for slots 7, 8, 9 and 10 (if the indexer is synchronized up to or past slot 10 and has at least 5 more deltas stored as events)
+Indexer 02: State at slot 6, Delta for slots 7 and 8 (if the indexer is only synchronised up to slot 8)
+Indexer 03: State at slot 8, Delta for slots 9 and 10 (if the indexer is synchronised up to or past slot 10, but only has events for slots 9 and 10 stored in the database)
+Indexer 04: No result (the indexer is not synchronised to slot 6)
+Best result (defined as the oldest available slot or all indexers), if calculated for indexers 1, 2 and 3 would be the data at slot 8. If we include the 4th indexer, then there is no result available as the 4th indexer is still synchronising.
+
+
+Argument
+--------
+
+A fairly simple solution to the synchronisation problem would be to qualify responses by the current slot number. This is a reasonable first thought, but it does have a high probability of returning results that are qualified by different slot numbers (if the indexers are fully synchronised, it would be a small offset, but still dependent on the various indexers processing speeds). The advantage of this solution is that at least, we now know that the indexers were not fully synchronised. If the application does not require them to be, then this is a reasonable solution.
```

Comment:
Similar to the comment about the semaphor, I would add here an argument explaining how the chosen approach is better. Might seem obvious, but it needs to be clearly stated.